bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/InactivityProfile.java

```diff
@@ -0,0 +1,157 @@
+    private final Cleaner.Cleanable cleanable;
+    private boolean closed;
+
+    private @Nullable ScheduledFuture<?> timeoutTask;
+
+    public InactivityProfile(ProfileCallback callback, ProfileContext context) {
+        InactivityProfileConfig config = context.getConfiguration().as(InactivityProfileConfig.class);
+        long mSec = 0;
+        try {
+            QuantityType<?> timeQty = QuantityType.valueOf(config.timeout);
+            if (!Units.SECOND.getDimension().equals(timeQty.getDimension())) {
+                throw new IllegalArgumentException();
+            }
+            mSec = timeQty.toUnit(MetricPrefix.MILLI(Units.SECOND)) instanceof QuantityType<?> mSecQty
+                    ? mSecQty.longValue()
+                    : 0;
+            if (mSec <= 0) {
+                throw new IllegalArgumentException();
+            }
+        } catch (IllegalArgumentException e) {
+            logger.warn("Profile configuration timeout value \"{}\" is invalid", config.timeout);
+        }
+
+        this.callback = callback;
+        this.scheduler = context.getExecutorService();
+        this.timeout = mSec > 0 ? Duration.ofMillis(mSec) : DEFAULT_TIMEOUT;
+        this.inverted = config.inverted != null ? config.inverted : false;
+        this.cleanable = CLEANER.register(this, () -> clean());
+
+        logger.debug("Created(timeout:{}, inverted:{})", timeout, inverted);
+
+        onStateUpdateFromHandler(UnDefType.NULL); // dummy to set initial item state
+    }
+
+    private void onTimeout() {
+        State itemState = OnOffType.from(!inverted);
+        logger.debug("timeout:{} => itemState:{}", timeout, itemState);
+        this.callback.sendUpdate(itemState);
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return INACTIVITY_UID;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State itemState) {
+        // do nothing
+    }
+
+    @Override
+    public void onCommandFromItem(Command itemCommand) {
+        // do nothing
+    }
+
+    @Override
+    public void onCommandFromHandler(Command handlerCommand) {
+        Command itemCommand = OnOffType.from(inverted);
+        logger.debug("handlerCommand:{} => itemCommand:{}", handlerCommand, itemCommand);
+        callback.sendCommand(itemCommand);
+        rescheduleTimeoutTask();
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State handlerState) {
+        State itemState = OnOffType.from(inverted);
+        logger.debug("handlerState:{} => itemState:{}", handlerState, itemState);
+        callback.sendUpdate(itemState);
+        rescheduleTimeoutTask();
+    }
+
+    @Override
+    public void close() throws Exception {
+        // causes immediate clean up instead of waiting on the garbage collector
+        cleanable.clean();
+    }
+
+    /**
+     * Called by the {@link Cleaner} callback
+     */
+    private void clean() {
+        cancelTimeoutTask();
+        closed = true;
+    }
```

Comment:
Can you add a trace log and see whether and when this actually gets called?

Or even better, try the same tests that I suggested in my [previous message](https://github.com/openhab/openhab-addons/pull/18301#discussion_r2160568385)

Reply:
I will add a Junit test